src/neptune/utils.py

```diff
@@ -73,3 +86,89 @@ def stop_synchronization_callback(neptune_object: NeptuneObject) -> None:
         "Threshold for disrupted synchronization exceeded. Stopping the synchronization using the default callback."
     )
     neptune_object.stop(seconds=DEFAULT_STOP_TIMEOUT)
+
+
+class NullProgressBar(ProgressBarCallback):
+    def __init__(self, *args: Any, description: Optional[str] = None, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        pass
+
+    def update(self, *, by: int, total: Optional[int] = None) -> None:
+        pass
+
+
+class TqdmProgressBar(ProgressBarCallback):
+    @handle_import_error(dependency="tqdm")
+    def __init__(
+        self,
+        *args: Any,
+        description: Optional[str] = None,
+        unit: Optional[str] = None,
+        unit_scale: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        from tqdm import tqdm
+
+        unit = unit if unit else ""
+
+        self._progress_bar = tqdm(desc=description, unit=unit, unit_scale=unit_scale, **kwargs)
+
+    def __enter__(self) -> "TqdmProgressBar":
+        self._progress_bar.__enter__()
+        return self
+
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc_val: Optional[BaseException],
+        exc_tb: Optional[TracebackType],
+    ) -> None:
+        self._progress_bar.__exit__(exc_type, exc_val, exc_tb)
+
+    def update(self, *, by: int, total: Optional[int] = None) -> None:
+        if total:
+            self._progress_bar.total = total
+        self._progress_bar.update(by)
+
+
+class TqdmNotebookProgressBar(ProgressBarCallback):
+    @handle_import_error(dependency="tqdm")
+    def __init__(
+        self,
+        *args: Any,
+        description: Optional[str] = None,
+        unit: Optional[str] = None,
+        unit_scale: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        from tqdm.notebook import tqdm
```

Comment:
This looks like the only difference between these two, could it be merged and just imports based on interactive vs non-interactive or something like that?